cmd/account/mgmt/account-unassign.go

```diff
@@ -128,57 +132,119 @@ func (o *accountUnassignOptions) run() error {
 		os.Exit(0)
 	}
 
-	// loop through accounts list and untag and move them back into root OU
 	for _, id := range accountIdList {
-
+		// untag account
 		err = o.untagAccount(id)
 		if err != nil {
 			return err
 		}
-
+		// move account
 		err = o.moveAccount(id, rootID, destinationOU)
 		if err != nil {
 			return err
 		}
+		// instantiate new client with AssumeRole
+		newAWSClient, err := o.assumeRoleForAccount(id)
+		if err != nil {
+			return err
+		}
+		// delete roles
+		err = o.deleteRoles(newAWSClient)
+		if err != nil {
+			return err
+		}
+		// list iam users created by each account and append to slice
+		users, err := o.listUsersFromAccount(newAWSClient, id)
+		if err != nil {
+			return err
+		}
+
+		allUsers = append(allUsers, users...)
+
 	}
 
-	// Delete login profile
-	err = o.deleteLoginProfile(accountUsername)
-	if err != nil {
-		return err
+	for _, userName := range allUsers {
+
+		// Delete login profile
+		err = o.deleteLoginProfile(userName)
```

Comment:
This could be simplified in a for loop that takes the fun too s as a param.and runs for each runs the check
This way was done in https://github.com/openshift/gcp-project-operator/blob/58ea50451037cda552783adc4b17b1300eba376c/pkg/controller/projectclaim/projectclaim_controller.go#L117